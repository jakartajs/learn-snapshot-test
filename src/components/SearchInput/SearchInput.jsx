import React, { PureComponent } from 'react';

import PropTypes from 'prop-types';

import Glyph from '../Glyph';
import styles from './styles.scss';

class SearchInput extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
    this.changeText = this.changeText.bind(this);
    this.search = this.search.bind(this);
  }

  changeText(e) {
    const { value } = e.target;

    this.setState({
      value
    });
  }

  search() {
    this.props.search(this.state.value);
  }

  render() {
    return (
      <div>
        <input
          style={{
            width: 200,
            height: 30,
            fontSize: 20
          }}
          value={this.state.value}
          onChange={this.changeText}
        />
        <button
          style={{
            width: 40,
            height: 35,
            fontSize: 20
          }}
          onClick={this.search}
        >
          <Glyph name={'search'} />
        </button>
      </div>
    );
  }
}

SearchInput.proptypes = {
  search: PropTypes.func.isRequired
};

export default SearchInput;
