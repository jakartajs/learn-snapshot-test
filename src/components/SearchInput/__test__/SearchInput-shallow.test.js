import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import React from 'react';
import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';

import SearchInput from '../SearchInput';

expect.addSnapshotSerializer(enzymeSerializer);

describe('SearchInput Shallow Component', () => {
  test('should have search input with empty search method', () => {
    const result = shallow(<SearchInput search={() => {}} />);
    expect(result).toMatchSnapshot();
  });

  test('should have search input with empty search method', () => {
    const result = shallow(<SearchInput search={() => {}} />);

    result.setState({
      value: '123'
    });
    expect(result).toMatchSnapshot();
  });
});
