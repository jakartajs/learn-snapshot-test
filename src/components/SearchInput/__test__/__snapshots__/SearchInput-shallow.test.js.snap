// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Shallow Component should have search input with empty search method 1`] = `
<div>
  <input
    onChange={[Function]}
    style={
      Object {
        "fontSize": 20,
        "height": 30,
        "width": 200,
      }
    }
    value=""
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "fontSize": 20,
        "height": 35,
        "width": 40,
      }
    }
  >
    <Glyph
      className=""
      name="search"
    />
  </button>
</div>
`;

exports[`SearchInput Shallow Component should have search input with empty search method 2`] = `
<div>
  <input
    onChange={[Function]}
    style={
      Object {
        "fontSize": 20,
        "height": 30,
        "width": 200,
      }
    }
    value="123"
  />
  <button
    onClick={[Function]}
    style={
      Object {
        "fontSize": 20,
        "height": 35,
        "width": 40,
      }
    }
  >
    <Glyph
      className=""
      name="search"
    />
  </button>
</div>
`;
